### DISCUSSION TITLE
Client-side optimisations when the site only needs server-side rendering

### DISCUSSION THREAD

**Discussion by philberesford** [vercel/next.js#12918] [Help] [answered]
Hi there!

This might be a very left-field requirement, but currently I'm working on a site that contains some pages that only require server-side rendering. 

For instance, some screens contain a HTML `<form>` and `<input type="submit">` buttons that must do a HTTP post to the server (like more traditional web 1.0 applications) without any JavaScript running in the browser. So, it would be nice to have the browser only load the HTML and CSS it needs for that initial paint, since we know the button press will result in a full post-back and page refresh anyway. This would save us several hundreds of kilobytes in the initial download from our servers.
 
Unfortunately, it appears that every page in the application also loads up JS chunks and framework code to the browser, despite the delivered content not needing any client-side functionality.

I've been through the documentation a few times and can't see any way of doing this whatsoever. Note that creating a static-site is also not an option since some data must be loaded up at run-time before the HTML is rendered. In these cases we're essentially using NextJS as a rich templating engine.

Is there a way that we can turn-off any client-side rendering, or only have server-side rendering enabled in a project, or even better within just some pages of a project please?

Any tips or suggestions would be greatly appreciated.
Thanks.

**Comment by timneutkens** [comment, marked as answer]
This is currently possible experimentally: https://github.com/zeit/next.js/pull/11949

**Comment by philberesford** [comment]
Fabulous! I'll take a look - and that was an incredibly quick reponse too. 'mazing.
Thank you.
